projects/plugins/jetpack/extensions/blocks/latex-formula/edit.tsx

```diff
@@ -0,0 +1,167 @@
+	const [ currentLatex, setCurrentLatex ] = useState< string >( latex );
+	const [ katexReady, setKatexReady ] = useState< boolean >( typeof window.katex !== 'undefined' );
+	const textareaRef = useRef< HTMLTextAreaElement >();
+	const props = useBlockProps( { className: 'jetpack-latex-formula is-editing' } );
+
+	useEffect( () => {
+		ensureKatexLoaded( () => setKatexReady( true ) );
+	}, [] );
+
+	useEffect( () => {
+		if ( editing && isSelected && textareaRef.current ) {
+			textareaRef.current.focus();
+		}
+	}, [ editing, isSelected ] );
+
+	useEffect( () => {
+		if ( ! katexReady ) {
+			setPreview( __( 'Loading formula preview…', 'jetpack' ) );
+			return;
+		}
+		try {
+			const html =
+				window.katex?.renderToString( currentLatex, {
+					throwOnError: false,
+					displayMode: true,
+				} ) || '';
+			setPreview( html );
+		} catch {
+			setPreview( __( 'Error rendering formula', 'jetpack' ) );
+		}
+	}, [ currentLatex, katexReady ] );
+
+	if ( editing ) {
+		return (
+			<div { ...props }>
+				<TextareaControl
+					label={ __( 'LaTeX Code', 'jetpack' ) }
+					ref={ textareaRef }
+					value={ currentLatex }
+					onChange={ ( next: string ) => setCurrentLatex( next ) }
+					help={ __( 'Enter LaTeX math to render below. Example: \\frac{a}{b}', 'jetpack' ) }
+					rows={ 4 }
+					autoFocus
+				/>
+				<div className="jetpack-latex-formula-live-preview">
+					{ katexReady ? (
+						<div
+							className="jetpack-latex-formula-rendered"
+							dangerouslySetInnerHTML={ { __html: preview } }
+						/>
+					) : (
+						<div className="jetpack-latex-formula-loading">
+							{ __( 'Loading formula preview…', 'jetpack' ) }
+						</div>
+					) }
+				</div>
+				<div className="jetpack-latex-formula-save-btn-row">
+					<Button
+						variant="primary"
+						onClick={ () => {
+							setAttributes( { latex: currentLatex, editing: false } );
+						} }
+						className="jetpack-latex-formula-save-btn"
+						disabled={ ! katexReady }
+					>
+						{ __( 'Done', 'jetpack' ) }
+					</Button>
+				</div>
+			</div>
+		);
+	}
+
+	// NOT editing: Immediately render the formula with KaTeX in-place (for a seamless editor experience)
+	return (
+		<div
+			{ ...useBlockProps( { className: 'jetpack-latex-formula' } ) }
+			tabIndex={ 0 }
+			role="button"
+			aria-label={ __( 'Edit LaTeX formula', 'jetpack' ) }
+			onClick={ () => setAttributes( { editing: true } ) }
+			data-latex={ latex }
+		>
+			{ katexReady ? (
+				<span
+					className="jetpack-latex-formula-rendered"
+					dangerouslySetInnerHTML={ {
+						__html: ( () => {
+							try {
+								return (
+									window.katex?.renderToString( latex, {
+										throwOnError: false,
+										displayMode: true,
+									} ) || ''
+								);
+							} catch {
+								return __( 'Error rendering formula', 'jetpack' );
+							}
+						} )(),
+					} }
+				/>
+			) : (
+				<span className="jetpack-latex-formula-loading" style={ { color: '#888' } }>
```

Comment:
Could use Loading animation here and move the colour to scss file: https://wordpress.github.io/gutenberg/?path=/docs/components-animate--docs#loading